docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,164 @@
+import org.partiql.spi.SourceLocation
+import org.partiql.spi.catalog.Session
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.PErrorKind
+import org.partiql.spi.errors.PErrorListener
+import org.partiql.spi.errors.PRuntimeException
+import org.partiql.spi.errors.Severity
+import org.partiql.spi.types.PType
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class ErrorHandlingChanges {
+
+    /**
+     * The example shows the default behavior of error handling in major PartiQL components {@link PartiQLParser},
+     * {@link PartiQLPlanner} and {@link PartiQLCompiler}. {@link PRuntimeException} will be thrown from the PartiQL system
+     * to trap any parsing errors or warnings. {@link PRuntimeException} wraps a {@link PError} which includes {@link PError#code()},
+     * {@link PError#kind}, {@link PError#severity}, {@link PError#location} and {@link PError#properties}.
+     * {@link PRuntimeException} with some enhanced information allows us for more robust and customized error control.
+     */
+    @Test
+    fun `Default PartiQL error handling`() {
+        val compiler = PartiQLCompiler.standard()
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val session = Session.empty()
+
+        val query = """
+            CREATE TABLE 'mytable;
+        """.trimIndent()
+
+        val exception = assertThrows<PRuntimeException> {
+            try {
+                val parseResult = parser.parse(query)
+                val plan = planner.plan(parseResult.statements[0], session).plan
+                val result = compiler.prepare(plan, Mode.STRICT())
+            } catch (e: PRuntimeException) {
+                // The exception should have been thrown from PartiQL components. You can add your handling here.
+                // Rethrow for validating the exception in test
+                throw e
+            }
+        }
+
+        assertEquals(exception.error.code(), PError.UNEXPECTED_TOKEN)
+        assertEquals(PErrorKind.SYNTAX, exception.error.kind.code())
+        assertEquals(Severity.ERROR, exception.error.severity.code())
+        assertEquals(SourceLocation(1, 14, 1), exception.error.location)
+        // Inspect properties from exception
+        assertEquals("UNRECOGNIZED", exception.error.get("TOKEN_NAME", String::class.java))
+    }
+
+    /**
+     * The example shows we registered a custom error handling in the context of major PartiQL components {@link PartiQLParser},
+     * {@link PartiQLPlanner} and {@link PartiQLCompiler} which records the errors or warning instead of throwing for deferred error processing.
+     * You can choose to treat warning as errors, or inhibit certain warnings or even ignore warnings.
+     * It might be a good idea for you to consolidate your logic of converting PErrors to user-presented messages and use that here.
+     */
+    @Test
+    fun `PartiQLParser with custom error handler`() {
```

Comment:
This example is demonstrating too many things IMO. We should break it up to cover a planner error example and have a separate example for a custom error handler.